notifications/management/commands/send_notifications.py

```diff
@@ -38,524 +22,53 @@ def add_arguments(self, parser):
-                              new_events=new_events)
-
-        if output is None:
-            self.stdout.write('no email')
-
-        else:
-            dthandler = lambda x: x.isoformat() if isinstance(x, date) else None
-            self.stdout.write(json.dumps(output, default=dthandler))
-
-
-    def find_bill_action_updates(self, bill_ids):
-
-        new_actions = '''
-            SELECT DISTINCT ON (bill.ocd_id)
-              bill.slug AS bill_slug,
-              bill.identifier AS bill_identifier,
-              bill.description AS bill_description,
-              action.description AS action_description,
-              action.date AS action_date
-            FROM new_action AS new
-            JOIN councilmatic_core_bill AS bill
-              ON new.bill_id = bill.ocd_id
-            JOIN councilmatic_core_action AS action
-              ON new.bill_id = action.bill_id
-            WHERE new.bill_id IN %s
-        '''
-
-        cursor = connection.cursor()
-        cursor.execute(new_actions, [tuple(bill_ids)])
-
-        bill_action_updates = []
-
-        for row in cursor:
-
-            bill = {
-                'slug': row[0],
-                'identifier': row[1],
-                'description': row[2],
-            }
-            action = {
-                'description': row[3],
-                'date': row[4],
-            }
-
-            bill_action_updates.append((bill, action))
-
-        return bill_action_updates
-
-    def find_bill_search_updates(self, search_params):
-
-        if not haystack_url:
-            self.stdout.write(self.style.ERROR('Solr is not configured so no search notifications will be sent'))
-            return []
-
-        search_updates = []
-
-        cursor = connection.cursor()
-
-        for params in search_params:
-
-            term = params['term'].strip()
-
-            if not term:
-                term = '*:*'
-
-            query_params = {
-                'q': term,
-                'fq': [],
-                'wt': 'json'
-            }
-
-            for facet, values in params['facets'].items():
-                for value in values:
-                    query_params['fq'].append('{0}:{1}'.format(facet, value))
-
-            results = requests.get('{}/select'.format(haystack_url), params=query_params)
-
-            ocd_ids = tuple(r['ocd_id'] for r in results.json()['response']['docs'])
-
-
-            new_bills = '''
-                SELECT DISTINCT ON (bill.ocd_id)
-                  bill.slug AS bill_slug,
-                  bill.identifier AS bill_identifier,
-                  bill.description AS bill_description
-                FROM new_bill AS new
-                JOIN councilmatic_core_bill AS bill
-                  ON new.ocd_id = bill.ocd_id
-                WHERE new.ocd_id IN %s
-            '''
-
-            if ocd_ids:
-                cursor.execute(new_bills, [ocd_ids])
-
-                bills = []
+    def _is_empty(self, elem):
```

Comment:
i'm slightly confused: wouldn't this return arrays that have one or more elements, i.e., arrays that are not empty?

also, since we only use this in the handle method, it might be a little cleaner to define it as a lambda function inside the handle method, but that's not a change i require!

Reply:
> wouldn't this return arrays that have one or more elements, i.e., arrays that are not empty?

Oops, true! Semantically, I was thinking `filter(self.is_empty, iterable)` would communicate that we're trying to remove (i.e. filter _out_) all empty objects from `iterable`, but the method itself is actually filtering _for_ non-empty objects. I'll change it to be something like `non_empty()`.

> since we only use this in the handle method, it might be a little cleaner to define it as a lambda function inside the handle method

Changing the `non_empty()` method to be local to the `handle()` method seems reasonable. I thought about assigning it to a lambda instead but it goes against the [PEP8 recommendation](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) to "Always use a def statement instead of an assignment statement that binds a lambda expression directly to an identifier." Still, we could do a one-line `def` statement and get the benefits of locality/brevity while not violating PEP8.

Reply:
oh good catch on pep, i didn't know that one! yeah, a one-line def would be rad.